src/internal_events/pulsar.rs

```diff
@@ -0,0 +1,38 @@
+use metrics::counter;
+use vector_core::internal_event::InternalEvent;
+
+use super::prelude::{error_stage, error_type};
+
+#[derive(Debug)]
+pub struct PulsarSendingError {
+    pub count: usize,
+    pub error: vector_core::Error,
+}
+
+impl InternalEvent for PulsarSendingError {
+    fn emit(self) {
+        error!(
+            message = "Events dropped.",
+            reason = "A Pulsar sink generated an error.",
+            error = %self.error,
+            error_type = error_type::REQUEST_FAILED,
+            intentional = "false",
```

Comment:
I thought we had changed these to booleans? Though I'm not sure it makes total sense, as we still need to convert them to strings for metrics.